detection/detection.go

```diff
@@ -0,0 +1,720 @@
+			log.Printf("Using native Go OpenCV for high-accuracy object detection")
+		} else {
+			log.Printf("Native OpenCV detection failed (%v), falling back to enhanced analysis", err)
+			// Fallback to original enhanced detection
+			result, err = analyzeWithEnhancedDetection(photoPath)
+		}
+	} else {
+		// Use enhanced detection directly
+		result, err = analyzeWithEnhancedDetection(photoPath)
+	}
+	
+	if err != nil {
+		return nil, err
+	}
+	
+	// Calculate detection latency
+	latency := time.Since(startTime)
+	result.LatencyMs = latency.Milliseconds()
+	
+	// Calculate overall confidence from individual detection confidence scores
+	result.OverallConfidence = calculateOverallConfidence(result.Details)
+	
+	// Log detection performance metrics
+	log.Printf("Object detection completed in %dms with overall confidence %.2f", 
+		result.LatencyMs, result.OverallConfidence)
+	
+	return result, nil
+}
+
+// DetectionConfig holds configuration for object detection
+type DetectionConfig struct {
+	UseOpenCV bool          // Whether to try OpenCV detection first
+	Timeout   time.Duration // Maximum time for detection
+}
+
+// analyzeWithEnhancedDetection performs the original enhanced detection as fallback
+func analyzeWithEnhancedDetection(photoPath string) (*DetectionResult, error) {
+	// Open and decode the image
+	file, err := os.Open(photoPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open photo: %v", err)
+	}
+	defer file.Close()
+
+	img, format, err := image.Decode(file)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode image: %v", err)
+	}
+
+	if format != "jpeg" && format != "png" && format != "gif" {
+		log.Printf("Warning: image format %s may not be fully supported, continuing with enhanced analysis", format)
+	}
+
+	// Initialize result
+	result := &DetectionResult{
+		PhotoPath:         photoPath,
+		Objects:           []string{},
+		Details:           []ObjectDetail{},
+		LatencyMs:         0, // Will be set by calling function
+		OverallConfidence: 0, // Will be calculated by calling function
+	}
+
+	// Analyze time of day using brightness
+	result.IsDay = analyzeTimeOfDayEnhanced(img)
+
+	// Perform enhanced object detection
+	objects, details := detectObjectsEnhanced(img)
+	result.Objects = objects
+	result.Details = details
+
+	// Generate summary
+	result.Summary = generateSummary(result.IsDay, objects)
+
+	return result, nil
+}
+
+// analyzeTimeOfDayEnhanced determines if it's day or night using enhanced brightness analysis
+func analyzeTimeOfDayEnhanced(img image.Image) bool {
+	bounds := img.Bounds()
+	width := bounds.Max.X - bounds.Min.X
+	height := bounds.Max.Y - bounds.Min.Y
+	
+	if width == 0 || height == 0 {
+		return true // default to day if we can't analyze
+	}
+
+	// Enhanced sampling with multiple metrics
+	sampleSize := 20 // More intensive sampling for better accuracy
+	totalBrightness := uint64(0)
+	totalContrast := uint64(0)
+	sampleCount := 0
```

Comment:
what is the expected accuracy of estimating day/night time purely from colors as done here compared to a opencv based solution